toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -10,10 +10,23 @@ import (
 	"strconv"
 
 	"github.com/microsoft/azurelinux/toolkit/tools/imagegen/diskutils"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/jsonutils"
 	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
 	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
 )
 
+type outputPartitionMetadata struct {
+	Type              string `json:"type"`         // Example: part
+	PartitionNum      int    `json:"partitionnum"` // Example: 1
+	PartitionFilepath string `json:"filepath"`     // Example: out/image_1.raw
```

Comment:
The `filepath` should probably be relative the `.json` file, not the working directory of the command prompt.